docs/backends/macos.rst

```diff
@@ -28,7 +28,31 @@ As stated above, this will however only work the macOS machine that performed
 the scan and thus cached the device as ``243E23AE-4A99-406C-B317-18F1BD7B4CBE``.
 
 There is also no pairing functionality implemented in macOS right now, since it does not seem
-to be any explicit pairing methods in the COre Bluetooth.
+to be any explicit pairing methods in the Core Bluetooth.
+
+CoreBluetooth does not differentiate between data from a notification, and data from a read.
+This can cause confusion in cases where a device may send a notification message on a characteristic
+that is an indication that the characteristic needs to be read again.
+
+Bleak has can accept a ``notification_discriminator`` callback in the ``cb`` dict parameter that is
+passed to the start_notify function that can differentiate between these types of data.
+
+.. code-block:: python
+
+    async def reread(char):
+        data = await client.read_gatt_char(char)
+
+    async def notification_handler(char, data):
+        asyncio.create_task(reread(char))
```

Comment:
This is buggy code. The returned task could be garbage-collected before completing. Also, it lacks any synchronization to prevent two reads at the same time. So best to not put that in an example that people might copy.

Instead, I would use an `asyncio.Event` and `set()` it here. Then in the main code after the `start_notify()`, have the following:

```python
while True:
    await event.wait()
    # we received a notification - prepare to receive another
    event.clear()
    # then read the characteristic to get the full value
    data = await client.read_gatt_char(char)
    # do stuff with data
```